src/OmenThumbnailMapping.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+interface IFixedProductMarketMaker {
+    function balanceOf(address account) external view returns (uint256);
+}
+
+contract OmenThumbnailMapping {
+    // Mapping from market's address to image's IPFS hash
+    mapping(address => string) private marketAddressToIPFSHash;
```

Comment:
The mapping should be `mapping(address => bytes32), see how Olas does it
https://gnosisscan.io/address/0x9338b5153AE39BB89f50468E608eD9d764B755fD#code#F1#L80

// Map of service Id => set of IPFS hashes pointing to the config metadata
    mapping (uint256 => bytes32[]) public mapConfigHashes;

Reason for that is this
https://ethereum.stackexchange.com/questions/11556/use-string-type-or-bytes32